cli/roger_push.py

```diff
@@ -385,19 +370,17 @@ def main(self, settings, appConfig, frameworkObject, hooksObj, args):
                     except Exception as e:
                         raise ValueError("Error while loading json from {} - {}".format(template_with_path, e))
 
-                    if '\"SECRET\"' in output:
-                        output = self.mergeSecrets(output, self.loadSecrets(
-                            secrets_dir, containerConfig, args, environment))
+                    if '\"SECRET\"' in output and not args.secrets_file:
+                        raise ValueError("Secrets present, need secret file to populate secrets")
+                    output = self.mergeSecrets(output, secret_vars)
                     if output != "StandardError":
                         try:
-                            comp_exists = os.path.exists("{0}".format(comp_dir))
-                            if comp_exists is False:
+                            comp_dir_exists = os.path.exists("{0}".format(comp_dir))
+                            if comp_dir_exists is False:
```

Comment:
if not comp_dir_exists:

On line 382 as well. Might as well fix the sillyness as we touch the code.